.github/workflows/test.yaml

```diff
@@ -31,4 +35,13 @@ jobs:
         if: always()
         with:
           name: Coverage results
-          path: ${{ steps.unity-test-step.outputs.coveragePath }}
\ No newline at end of file
+          path: ${{ steps.unity-test-step.outputs.coveragePath }}
+      - name: SonarCloud Scan
+        uses: sonarsource/sonarcloud-github-action@v2.2.0
+        env:
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
```

Comment:
Where is the secret stored? If it is a repository-wide secret, this will fail when ran from a forked PR

Reply:
Yes, on both repos this has always been a repo wide secret, this is normal in open source development. It is a secret, we don't want to expose it to outside sources.

Reply:
That's correct, but I believe there are ways to provide some credentials for these type of pipelines which can be accessed in a forked PR. We should consider this, for if someone happily helps us, he gets greeted with "oh sorry the pipeline doesn't work yet"

Reply:
I don't think we want to expose secrets to outside sources at all

Reply:
It is quite normal for open source repositories to enforce people run pipelines on their own, and linking to that when opening the PR, this is what I have encountered multiple times, one of which is when I contributed to game-ci (the org that created/s the actions for unity runners, testers and builders for pipelines.

Reply:
How can they run it on their own if they do not have the secret?

Reply:
Discord me about this :)

Reply:
Let's keep all PR related conversations in GitHub 😄 This way, other contributors or reviewers are kept up-to-date with the progress of the discussion.

Reply:
We can include our conclusion, but this communication is way too slow

Reply:
> How can they run it on their own if they do not have the secret?

They can't, they need to set up their own, that is normal for open source development, then when it has ran on their fork, they include a link to the action logs and result (and in this case sonarcloud link) within the PR description.

Reply:
Bruh